examples/src/main/java/com/mapbox/navigation/examples/core/InstructionViewActivity.kt

```diff
@@ -0,0 +1,429 @@
+        val requestLocationUpdateRequest =
+            LocationEngineRequest.Builder(1000L)
+                .setPriority(LocationEngineRequest.PRIORITY_NO_POWER)
+                .build()
+
+        locationEngine.requestLocationUpdates(
+            requestLocationUpdateRequest,
+            locationListenerCallback,
+            mainLooper
+        )
+        locationEngine.getLastLocation(locationListenerCallback)
+    }
+
+    private fun stopLocationUpdates() {
+        locationEngine.removeLocationUpdates(locationListenerCallback)
+    }
+
+    private fun showFeedbackBottomSheet() {
+        supportFragmentManager?.let {
+            FeedbackBottomSheet.newInstance(
+                this,
+                NavigationConstants.FEEDBACK_BOTTOM_SHEET_DURATION
+            )
+                .show(it, FeedbackBottomSheet.TAG)
+        }
+    }
+
+    private fun encodeSnapshot(snapshot: Bitmap): String {
+        screenshotView.visibility = VISIBLE
+        screenshotView.setImageBitmap(snapshot)
+        mapView.visibility = View.INVISIBLE
+        val encodedSnapshot = ViewUtils.encodeView(ViewUtils.captureView(mapView))
+        screenshotView.visibility = View.INVISIBLE
+        mapView.visibility = VISIBLE
+        return encodedSnapshot
+    }
+
+    private fun initViews() {
+        startNavigation.visibility = VISIBLE
+        startNavigation.isEnabled = false
+        instructionView.visibility = GONE
+        feedbackButton = instructionView.retrieveFeedbackButton().apply {
+            hide()
+            addOnClickListener {
+                showFeedbackBottomSheet()
+            }
+        }
+        instructionSoundButton = instructionView.retrieveSoundButton().apply {
+            hide()
+            addOnClickListener {
+                val soundButton = instructionSoundButton
+                if (soundButton is SoundButton) {
+                    speechPlayer.isMuted = soundButton.toggleMute()
+                }
+            }
+        }
+    }
+
+    private fun updateViews(tripSessionState: TripSessionState) {
+        when (tripSessionState) {
+            TripSessionState.STARTED -> {
+                startNavigation.visibility = GONE
+                instructionView.visibility = VISIBLE
+                feedbackButton?.show()
+                instructionSoundButton?.show()
+            }
+            TripSessionState.STOPPED -> {
+                startNavigation.visibility = VISIBLE
+                startNavigation.isEnabled = false
+                instructionView.visibility = GONE
+                feedbackButton?.hide()
+                instructionSoundButton?.hide()
+            }
+        }
+    }
+
+    // Callbacks and Observers
+    private val routesReqCallback = object : RoutesRequestCallback {
+        override fun onRoutesReady(routes: List<DirectionsRoute>) {
+            Timber.d("route request success %s", routes.toString())
+            if (routes.isNotEmpty()) {
+                route = routes[0]
```

Comment:
We can add the line here `replayRouteLocationEngine.assign(routes[0])`

Reply:
yes we can. May I know is there any specific reason that prefer this way over the current implementation? I mean if it doesn't have problem, then I prefer leave it as is.

Reply:
Usually, we should not maintain state if possible. This particular state is of no use other than assigning it to `replayRouteLocationEngine`. It can always be done inside `onRoutesReady()`